Add "IN" conditions in data filter

melohagan opened the pull request:
## Description
You can now filter to check if a value is contained within a provided array.
*The "NOT IN" case is not done as part of this PR*

Addresses: 
- https://github.com/Budibase/budibase/issues/3863

## Screenshots
![Screenshot 2022-06-24 at 14 45 57](https://user-images.githubusercontent.com/101575380/175548841-a4cc478b-486e-4d02-adc9-5dc1efb8e340.png)

**Result**
![Screenshot 2022-06-24 at 14 48 12](https://user-images.githubusercontent.com/101575380/175549232-7dc30691-34da-48a2-b4e3-e210fb092d27.png)

aptkingston commented:
What's the support like for this Mel? I didn't realise `oneOf` and `notOneOf` were valid lucene operators - pretty cool! I can see there's a client side implementation so I'm assuming it works for custom queries, but have you also confirmed it works for internal tables and SQL tables?

melohagan commented:
Ah @aptkingston I've misunderstood how this works. I thought all filters went through this client side code, but it makes sense that it would do the filtering server side for SQL etc!

With that said, by sheer fluke, *oneOf* does work for SQL 😂 

**Postgres**
![Screenshot 2022-06-24 at 16 39 46](https://user-images.githubusercontent.com/101575380/175569705-4f529760-9041-4e50-8ef1-70c6e50fe151.png)
![Screenshot 2022-06-24 at 16 40 59](https://user-images.githubusercontent.com/101575380/175569925-17870c84-146b-4d08-8ecc-4d4fdf5b76be.png)
**JS**: `return [ "Apple", "Banana", "Carrot" ]`
 
**MySQL**
![Screenshot 2022-06-24 at 16 44 22](https://user-images.githubusercontent.com/101575380/175570509-0c6d251a-33bf-4527-bd8a-6193688fd271.png)
![Screenshot 2022-06-24 at 16 45 23](https://user-images.githubusercontent.com/101575380/175570707-a36c115d-847d-40a7-827c-db32f0a0b1ee.png)
**JS**: `return [ "11.3453", "234", "line1\nline2\nline3" ]`

----
**notOneOf** is not valid at all, so I could remove it for the time being as I don't need it. Looking to use this for handling relationships from Airtable queries - but it's been requested as well of course. 

**oneOf** is also not working for Internal Tables, but looking at `internalSearch.js` hopefully I can figure out an implementation there.

melohagan commented:
@aptkingston I've added support for Internal tables. From what I could find online, the only way to do an "IN" with Lucene queries is to concatenate OR statements. Apparently there's a cap of 1024 ORs by default - it'll be up to the user to use this within reason. I can slip that somewhere into the docs.

I also removed the **Is not in** filter option, but the **Is in** should now work for internal searches, external searches and custom queries.

aptkingston commented on packages/frontend-core/src/utils/lucene.js:
```diff
@@ -211,6 +212,11 @@ export const runLuceneQuery = (docs, query) => {
 
+  // Process an includes match (fails if the value is not included)
+  const oneOf = match("oneOf", (docValue, testValue) => {
+    return !testValue?.includes(docValue)
```
Looks great @melohagan! Could we copy the logic here to support splitting strings into arrays if the value is a string? The same way I see that it gets handled server side. Would be nice to keep support consistent, otherwise comma separated strings will work with internal/SQL but not with custom queries.

aptkingston reviewed (approved):
LGTM!
